src/sponsorship.js

```diff
@@ -1,42 +1,83 @@
 // FIXME: how should we remove the `dist/` from this path?
 // NOTE: it requires this PR to work: https://github.com/readthedocs/ethical-ad-client/pull/148
 import { load_placements } from "ethical-ad-client/dist/ethicalads";
+import { AddonBase } from "./utils";
 
 const EXPLICIT_PLACEMENT_SELECTOR = "[data-ea-publisher]";
 
-function createAdPlacement() {
-  let placement;
-
-  placement = document.querySelector(EXPLICIT_PLACEMENT_SELECTOR);
-  if (placement) {
-    placement.setAttribute("data-ea-publisher", "readthedocs");
-    placement.setAttribute("data-ea-manual", "true");
-    if (
-      placement.getAttribute("data-ea-type") !== "image" &&
-      placement.getAttribute("data-ea-type") !== "text"
-    ) {
-      placement.setAttribute("data-ea-type", "readthedocs-sidebar");
+/**
+ * Sponsoship addon
+ *
+ * Show an Ad in the documentation page.
+ *
+ * Read more at:
+ *  - https://docs.readthedocs.io/en/stable/advertising/ethical-advertising.html
+ *
+ * @param {Object} config - Addon configuration object
+ */
+export class SponsorshipAddon extends AddonBase {
```

Comment:
Same question here on naming, but otherwise:

```suggestion
/**
 * Sponsorship addon
 *
 * Show an ad in the documentation page.
 *
 * Read more at:
 *  - https://docs.readthedocs.io/en/stable/advertising/ethical-advertising.html
 *
 * @param {Object} config - Addon configuration object
 */
export class SponsorshipAddon extends AddonBase {
```